great_expectations/checkpoint/actions.py

```diff
@@ -973,22 +947,24 @@ class UpdateDataDocsAction(ValidationAction):
     ```
 
     Args:
-        data_context: Data Context that is used by the Action.
         site_names: Optional. A list of the names of sites to update.
     """
 
     def __init__(
         self,
-        data_context: AbstractDataContext,
-        site_names: list[str] | str | None = None,
+        site_names: list[str] | None = None,
     ) -> None:
         """
-        :param data_context: Data Context
         :param site_names: *optional* List of site names for building data docs
         """
-        super().__init__(data_context)
         self._site_names = site_names
 
+        from great_expectations import project_manager
```

Comment:
I may consider moving this to the base class and provide one context API for all actions. I'm not sure what is better.
If the `project_manager` is internal to GX but we expect people to contribute `actions` then this would be a way to provide actions an API for these functions without making the project_manager public. If however, we don't expect anyone to contribute actions or if we expect contributed actions to not care about these things, we could do it this way.

Reply:
I'd vote for moving to the base class. Actions seem like a nice way for users to contribute.